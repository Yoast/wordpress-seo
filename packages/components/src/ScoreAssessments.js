import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const ScoreAssessmentItem = styled.li`
	display: table-row;
	font-size: 14px;
`;

const ScoreAssessmentPart = styled.span`
	display: table-cell;
	padding: 2px;
`;

const ScoreAssessmentBullet = styled( ScoreAssessmentPart )`
	position: relative;
	top: 1px;
	display: inline-block;
	height: 8px;
	width: 8px;
	border-radius: 50%;
	background-color: ${ props => props.scoreColor };
`;

ScoreAssessmentBullet.propTypes = {
	scoreColor: PropTypes.string.isRequired,
};

const ScoreAssessmentText = styled( ScoreAssessmentPart )`
	padding-left: 8px;
	width: 100%;
`;

const ScoreAssessmentScore = styled( ScoreAssessmentPart )`
	font-weight: 600;
	text-align: right;
	padding-left: 16px;
`;

/**
 * The ScoreAssessmentItem component, with a bullet, text and score.
 *
 * @param {object} props The component's props.
 *
 * @returns {ReactElement} The ScoreAssessmentItem component.
 */
const ScoreAssessment = ( props ) => {
	return (
		<ScoreAssessmentItem
			className={ `${ props.className }` }
		>
			<ScoreAssessmentBullet
				className={ `${ props.className }-bullet` }
				scoreColor={ props.scoreColor }
			/>
			<ScoreAssessmentText
				className={ `${ props.className }-text` }
				dangerouslySetInnerHTML={ { __html: props.html } }
			/>
			{ props.value &&
				<ScoreAssessmentScore
					className={ `${ props.className }-score` }
				>
					{ props.value }
				</ScoreAssessmentScore>
			}
		</ScoreAssessmentItem>
	);
};

ScoreAssessment.propTypes = {
	className: PropTypes.string.isRequired,
	scoreColor: PropTypes.string.isRequired,
	html: PropTypes.string.isRequired,
	value: PropTypes.number,
};

const ScoreAssessmentList = styled.ul`
	display: table;
	box-sizing: border-box;
	list-style: none;
	max-width: 100%;
	min-width: 200px;
	margin: 8px 0;
	padding: 0 8px;
`;

const ScoreAssessments = ( props ) => {
	return (
		<ScoreAssessmentList
			className={ props.className }
			role="list"
		>
			{ props.items.map( ( item, index ) =>
				<ScoreAssessment
					className={ `${ props.className }__item` }
					key={ index }
					scoreColor={ item.color }
					html={ item.html }
					value={ item.value }
				/>
			) }
		</ScoreAssessmentList>
	);
};

ScoreAssessments.propTypes = {
	className: PropTypes.string,
	items: PropTypes.arrayOf(
		PropTypes.shape( {
			color: PropTypes.string.isRequired,
			html: PropTypes.string.isRequired,
			value: PropTypes.number,
		} )
	),
};

ScoreAssessments.defaultProps = {
	className: "score-assessments",
};

export default ScoreAssessments;
