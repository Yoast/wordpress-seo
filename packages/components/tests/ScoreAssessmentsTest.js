import React from "react";
import renderer from "react-test-renderer";

import ScoreAssessments from "../src/ScoreAssessments";

test( "the ScoreAssessments matches the snapshot", () => {
	const items = [
		{
			color: "#F00",
			score: "20",
			html: "<a href=\"http://google.com\">Posts with a <strong>bad</strong> SEO score</a>:",
		},
		{
			color: "#0F0",
			score: "50",
			html: "<a href=\"http://google.com\">Posts with an <strong>OK</strong> SEO score</a>:",
		},
	];

	const component = renderer.create(
		<ScoreAssessments items={ items } />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the ScoreAssessments without scores matches the snapshot", () => {
	const items = [
		{
			color: "#F00",
			html: "<a href=\"http://google.com\">Posts with a <strong>bad</strong> SEO score</a>:",
		},
		{
			color: "#0F0",
			html: "<a href=\"http://google.com\">Posts with an <strong>OK</strong> SEO score</a>:",
		},
	];

	const component = renderer.create(
		<ScoreAssessments items={ items } />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the ScoreAssessments with some scores matches the snapshot", () => {
	const items = [
		{
			color: "#F00",
			score: "20",
			html: "<a href=\"http://google.com\">Posts with a <strong>bad</strong> SEO score</a>:",
		},
		{
			color: "#0F0",
			html: "<a href=\"http://google.com\">Posts with an <strong>OK</strong> SEO score</a>:",
		},
	];

	const component = renderer.create(
		<ScoreAssessments items={ items } />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
