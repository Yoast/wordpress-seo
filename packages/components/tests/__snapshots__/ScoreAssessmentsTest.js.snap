// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ScoreAssessments matches the snapshot 1`] = `
.c1 {
  display: table-row;
  font-size: 14px;
}

.c2 {
  display: table-cell;
  padding: 2px;
}

.c3 {
  position: relative;
  top: 1px;
  display: inline-block;
  height: 8px;
  width: 8px;
  border-radius: 50%;
  background-color: #F00;
}

.c5 {
  position: relative;
  top: 1px;
  display: inline-block;
  height: 8px;
  width: 8px;
  border-radius: 50%;
  background-color: #0F0;
}

.c4 {
  padding-left: 8px;
  width: 100%;
}

.c0 {
  display: table;
  box-sizing: border-box;
  list-style: none;
  max-width: 100%;
  min-width: 200px;
  margin: 8px 0;
  padding: 0 8px;
}

<ul
  className="c0 score-assessments"
  role="list"
>
  <li
    className="c1 score-assessments__item"
  >
    <span
      className="c2 c3 score-assessments__item-bullet"
    />
    <span
      className="c2 c4 score-assessments__item-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<a href=\\"http://google.com\\">Posts with a <strong>bad</strong> SEO score</a>:",
        }
      }
    />
  </li>
  <li
    className="c1 score-assessments__item"
  >
    <span
      className="c2 c5 score-assessments__item-bullet"
    />
    <span
      className="c2 c4 score-assessments__item-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<a href=\\"http://google.com\\">Posts with an <strong>OK</strong> SEO score</a>:",
        }
      }
    />
  </li>
</ul>
`;

exports[`the ScoreAssessments with some scores matches the snapshot 1`] = `
.c1 {
  display: table-row;
  font-size: 14px;
}

.c2 {
  display: table-cell;
  padding: 2px;
}

.c3 {
  position: relative;
  top: 1px;
  display: inline-block;
  height: 8px;
  width: 8px;
  border-radius: 50%;
  background-color: #F00;
}

.c5 {
  position: relative;
  top: 1px;
  display: inline-block;
  height: 8px;
  width: 8px;
  border-radius: 50%;
  background-color: #0F0;
}

.c4 {
  padding-left: 8px;
  width: 100%;
}

.c0 {
  display: table;
  box-sizing: border-box;
  list-style: none;
  max-width: 100%;
  min-width: 200px;
  margin: 8px 0;
  padding: 0 8px;
}

<ul
  className="c0 score-assessments"
  role="list"
>
  <li
    className="c1 score-assessments__item"
  >
    <span
      className="c2 c3 score-assessments__item-bullet"
    />
    <span
      className="c2 c4 score-assessments__item-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<a href=\\"http://google.com\\">Posts with a <strong>bad</strong> SEO score</a>:",
        }
      }
    />
  </li>
  <li
    className="c1 score-assessments__item"
  >
    <span
      className="c2 c5 score-assessments__item-bullet"
    />
    <span
      className="c2 c4 score-assessments__item-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<a href=\\"http://google.com\\">Posts with an <strong>OK</strong> SEO score</a>:",
        }
      }
    />
  </li>
</ul>
`;

exports[`the ScoreAssessments without scores matches the snapshot 1`] = `
.c1 {
  display: table-row;
  font-size: 14px;
}

.c2 {
  display: table-cell;
  padding: 2px;
}

.c3 {
  position: relative;
  top: 1px;
  display: inline-block;
  height: 8px;
  width: 8px;
  border-radius: 50%;
  background-color: #F00;
}

.c5 {
  position: relative;
  top: 1px;
  display: inline-block;
  height: 8px;
  width: 8px;
  border-radius: 50%;
  background-color: #0F0;
}

.c4 {
  padding-left: 8px;
  width: 100%;
}

.c0 {
  display: table;
  box-sizing: border-box;
  list-style: none;
  max-width: 100%;
  min-width: 200px;
  margin: 8px 0;
  padding: 0 8px;
}

<ul
  className="c0 score-assessments"
  role="list"
>
  <li
    className="c1 score-assessments__item"
  >
    <span
      className="c2 c3 score-assessments__item-bullet"
    />
    <span
      className="c2 c4 score-assessments__item-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<a href=\\"http://google.com\\">Posts with a <strong>bad</strong> SEO score</a>:",
        }
      }
    />
  </li>
  <li
    className="c1 score-assessments__item"
  >
    <span
      className="c2 c5 score-assessments__item-bullet"
    />
    <span
      className="c2 c4 score-assessments__item-text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<a href=\\"http://google.com\\">Posts with an <strong>OK</strong> SEO score</a>:",
        }
      }
    />
  </li>
</ul>
`;
